pkg/api/v1/schema.go

```diff
@@ -210,12 +210,33 @@ type CPU struct {
 	// Defaults to host-model.
 	// +optional
 	Model string `json:"model,omitempty"`
+	// Features specifies the CPU features list inside the VMI.
+	// For more information see https://libvirt.org/formatdomain.html#elementsCPU.
+	// +optional
+	Features []CPUFeature `json:"features,omitempty"`
 	// DedicatedCPUPlacement requests the scheduler to place the VirtualMachineInstance on a node
 	// with enough dedicated pCPUs and pin the vCPUs to it.
 	// +optional
 	DedicatedCPUPlacement bool `json:"dedicatedCpuPlacement,omitempty"`
 }
 
+// CPUFeature allows specifying a CPU feature.
+// ---
+// +k8s:openapi-gen=true
+type CPUFeature struct {
+	// Name of the CPU feature
+	Name string `json:"name"`
+	// Policy is the CPU feature attribute which can have the following attributes:
+	// force    - The virtual CPU will claim the feature is supported regardless of it being supported by host CPU.
+	// require  - Guest creation will fail unless the feature is supported by the host CPU or the hypervisor is able to emulate it.
+	// optional - The feature will be supported by virtual CPU if and only if it is supported by host CPU.
+	// disable  - The feature will not be supported by virtual CPU.
+	// forbid   - Guest creation will fail if the feature is supported by host CPU.
+	// Defaults to require
```

Comment:
It would be nice to indicate how each policy value impacts scheduling here.

Reply:
Im not sure this is the right place for that.
The explanation for the schema here will be overloaded with specific information because the scheduling decisions made here for forbid/require/non are taken place in case the NFD feature is enabled and the nodes have the matching labels.